datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -1082,13 +1083,16 @@ mod test {
 
     #[test]
     fn binary_op_date32_op_interval() -> Result<()> {
-        //CAST(Utf8("1998-03-18") AS Date32) + IntervalDayTime("386547056640")
+        // CAST(Utf8("1998-03-18") AS Date32) + IntervalDayTime("...")
         let expr = cast(lit("1998-03-18"), DataType::Date32)
-            + lit(ScalarValue::IntervalDayTime(Some(386547056640)));
+            + lit(ScalarValue::IntervalDayTime(Some(IntervalDayTime {
+                days: 123,
+                milliseconds: 456,
+            })));
```

Comment:
BTW I think we could also use 

https://github.com/apache/datafusion/blob/6846513616608fa34e2aa92495df9f879bbd063a/datafusion/common/src/scalar/mod.rs#L942-L945

to reduce the boilerplate:

```suggestion
            + lit(ScalarValue::new_interval_dt(123,  456))
```

But it is less explicit